ocaml/idl/datamodel.ml

```diff
@@ -3891,16 +3892,120 @@ let host_crashdump =
+  ~forward_to:(HostExtension "Pool_update.precheck")
+  ()
+
+let pool_update_apply = call
+  ~name:"apply"
+  ~doc:"Apply the selected update to a host and return its output"
+  ~in_oss_since:None
+  ~in_product_since:rel_ely
+  ~params:[ Ref _pool_update, "self", "The update to apply"; Ref _host, "host", "The host to apply the update too" ]
+  ~allowed_roles:_R_POOL_OP
+  ~forward_to:(HostExtension "Pool_update.apply")
+  ()
+
+let pool_update_pool_apply = call
+  ~name:"pool_apply"
+  ~doc:"Apply the selected update to all hosts in the pool and return a map of host_ref -> update output"
+  ~in_oss_since:None
+  ~in_product_since:rel_ely
+  ~params:[ Ref _pool_update, "self", "The update to apply"]
+  ~allowed_roles:_R_POOL_OP
+  ()
+
+let pool_update_clean = call
+  ~name:"clean"
+  ~doc:"Removes the update's files from the server"
+  ~in_oss_since:None
+  ~in_product_since:rel_ely
+  ~params:[ Ref _pool_update, "self", "The update to clean up"; Ref _host, "host", "The host to clean the update too" ] 
+  ~allowed_roles:_R_POOL_OP
+  ()
+
+let pool_update_pool_clean = call
+  ~name:"pool_clean"
+  ~doc:"Removes the update's files from all hosts in the pool, but does not remove the database entries"
+  ~in_oss_since:None
+  ~in_product_since:rel_ely
+  ~params:[ Ref _pool_update, "self", "The update to clean up" ]
+  ~allowed_roles:_R_POOL_OP
+  ()
+
+let pool_update_destroy = call
+  ~name:"destroy"
+  ~doc:"Removes the update's files from all hosts in the pool, and removes the database entries.  Only works on unapplied updates."
+  ~in_oss_since:None
+  ~in_product_since:rel_ely
+  ~params:[ Ref _pool_update, "self", "The update to destroy" ]
+  ~allowed_roles:_R_POOL_OP
+  ()
+
+let pool_update =
+  create_obj ~in_db:true 
+    ~in_product_since:rel_ely 
+    ~in_oss_since:None 
+    ~internal_deprecated_since:None
+
+    ~persist:PersistEverything 
+    ~gen_constructor_destructor:false 
+    ~gen_events:true
+
+    ~name:_pool_update 
+    ~descr:"Pool-wide updates"
+    ~doccomments:[]
+    ~messages_default_allowed_roles:_R_POOL_OP
+    ~messages:[
+    	pool_update_introduce;
+    	pool_update_precheck;
+    	pool_update_apply;
+    	pool_update_pool_apply;
+    	pool_update_clean;
+    	pool_update_pool_clean;
+    	pool_update_destroy;
+    ]
+    ~contents:
+    [ uid       ~in_oss_since:None _pool_update;
+      namespace ~name:"name" ~contents:(names None StaticRO) ();
+      field     ~in_product_since:rel_ely ~default_value:(Some (VInt Int64.zero)) ~in_oss_since:None ~qualifier:DynamicRO ~ty:Int "installation_size" "Size of the update in bytes";
+      field     ~in_product_since:rel_ely ~default_value:(Some (VSet [])) ~in_oss_since:None ~qualifier:DynamicRO ~ty:(Set pool_update_after_apply_guidance) "after_apply_guidance" "What the client should do after this update has been applied.";
+      field     ~in_product_since:rel_ely ~in_oss_since:None ~qualifier:DynamicRO ~ty:(Ref _vdi) "vdi" "VDI the update was uploaded to";
```

Comment:
Thanks, updated.
But meanwhile, seems that we must remove `~in_product_since:rel_ely` for this field.
Please correct it any misunderstanding.